### PR TITLE
Mark client as QDELING

### PR DESCRIPTION
**Pull request by Mothblocks**
Clients will now be marked QDELING while in the process of deleting. Necessary for #75445

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -536,6 +536,10 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 	return ..()
 
 /client/Destroy()
+	gc_destroyed = world.time
+	if (!QDELING(src))
```
Comment (Cyberboss) [inline code/modules/client/client_procs.dm:540]: Isn't this redundant? Only time it'd fail is if `world.time == 0` and it should be impossible for clients to connect then.

IIRC the implementation of `QDELING(X)` is `(##X.gc_destroyed)`

Comment (Mothblocks) [reply code/modules/client/client_procs.dm:540]: This serves two purposes:

1. It acts as an explanation for the line above it
2. If `QDELING` definition changes (which it definitely reasonably could), we will know if clients no longer get marked as `QDELING`, something we can easily do in a test.

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -536,6 +536,10 @@ GLOBAL_LIST_INIT(blacklisted_builds, list(
 	return ..()
 
 /client/Destroy()
+	gc_destroyed = world.time
```
Comment (LemonInTheDark) [inline code/modules/client/client_procs.dm:539]: why here and not Del?

Comment (Mothblocks) [reply code/modules/client/client_procs.dm:539]: I'll move it

Comment (Mothblocks) [reply code/modules/client/client_procs.dm:539]: Actually how much does this matter, should we call this before the send signal?

image

Comment (LemonInTheDark) [reply code/modules/client/client_procs.dm:539]: ... yeah, I think we should. that's how qdel() does it at least

**Pull request merged**